### PR TITLE
Update to AFL 2.52b

### PR DESCRIPTION
**Pull request by anon8675309**
Get ready for an incredible contribution:
Resolves https://github.com/shellphish/shellphish-afl/issues/8 :-)

Will also require a PR to shellphish-afl to patch up the eflload patch.

### CODE REVIEW

**Comment by zardus** [comment]
Thank you for this PR!

Rather than merge the new version into the old, could you rebase the repo on top of the new version? Similar to https://github.com/shellphish/afl-other-arch/pull/2.

Thank you!

**Comment by anon8675309** [comment]
OK, we can do that if that's what it takes to get accepted, but why on Earth would you want that?

This does, however, explain why we got merge conflicts out of nowhere when we were trying to submit our PR.

**Comment by zardus** [comment]
We have a few commits to adapt AFL, and they're done on top of the AFL source code. If those commits are rebased when each new version of AFL is released, then the commit log makes sense. Otherwise, it becomes a nightmarish graph of merge conflict resolutions, and our actual modifications are lost in the chaos.

The current graph of commits shows the optimal solution: https://github.com/shellphish/afl-other-arch/commits/master

Basically, best-case scenario, the "update AFL to 2.52b" commit would go between 9263d4599e7b704bf4b147085143ced06567c526 and d04ba735110318d42ca646d56d533f9202bfb26d.

**Comment by anon8675309** [comment]
New pull request submitted.  It won't merge cleanly, but I think this is what you wanted.

**Comment by bannsec** [comment]
Was actually going to look into this exact thing. What needs to be done to get this updated to a non-archaic version of afl?

**Comment by zardus** [comment]
Years later, this can be closed! This repo has been obsoleted by build process updates in shellphish-afl.
